src/absil/ilwritepdb.fs

```diff
@@ -0,0 +1,634 @@
+    serializer.SerializeMetadata(blobBuilder, (fun builder -> pdbContentId)) |> ignore
+ 
+    reportTime showTimes "PDB: Created"
+    use portablePdbStream = new FileStream(fpdb, FileMode.Create, FileAccess.ReadWrite)
+    blobBuilder.WriteContentTo(portablePdbStream)
+    reportTime showTimes "PDB: Closed"
+    pdbGetDebugInfo info.ModuleID fpdb
+
+#if FX_NO_PDB_WRITER
+#else
+//---------------------------------------------------------------------
+// PDB Writer.  The function [WritePdbInfo] abstracts the 
+// imperative calls to the Symbol Writer API.
+//---------------------------------------------------------------------
+let WritePdbInfo fixupOverlappingSequencePoints showTimes f fpdb info = 
+    try FileSystem.FileDelete fpdb with _ -> ()
+    let pdbw = ref Unchecked.defaultof<PdbWriter>
+    
+    try
+        pdbw := pdbInitialize f fpdb
+    with _ -> error(Error(FSComp.SR.ilwriteErrorCreatingPdb(fpdb), rangeCmdArgs))
+
+    match info.EntryPoint with 
+    | None -> () 
+    | Some x -> pdbSetUserEntryPoint !pdbw x 
+
+    let docs = info.Documents |> Array.map (fun doc -> pdbDefineDocument !pdbw doc.File)
+    let getDocument i = 
+      if i < 0 || i > docs.Length then failwith "getDocument: bad doc number"
+      docs.[i]
+    reportTime showTimes (sprintf "PDB: Defined %d documents" info.Documents.Length)
+    Array.sortInPlaceBy (fun x -> x.MethToken) info.Methods
+    reportTime showTimes (sprintf "PDB: Sorted %d methods" info.Methods.Length)
+
+    // This next bit is a workaround.  The sequence points we get 
+    // from F# (which has nothing to do with this module) are actually expression 
+    // marks, i.e. the source ranges they denote are typically 
+    // nested, and each point indicates where the  
+    // code for an expression with a particular range begins.  
+    // This is in many ways a much more convenient form to emit. 
+    // However, it is not the form that debug tools accept nicely. 
+    // However, sequence points are really a non-overlapping, non-nested 
+    // partition of the source code of a method.  So here we shorten the 
+    // length of all sequence point marks so they do not go further than 
+    // the next sequence point in the source. 
+    let spCounts =  info.Methods |> Array.map (fun x -> x.SequencePoints.Length)
+    let allSps = Array.concat (Array.map (fun x -> x.SequencePoints) info.Methods |> Array.toList)
+    let allSps = Array.mapi (fun i sp -> (i,sp)) allSps
+    if fixupOverlappingSequencePoints then 
+        // sort the sequence points into source order 
+        Array.sortInPlaceWith (fun (_,sp1) (_,sp2) -> SequencePoint.orderBySource sp1 sp2) allSps
+        // shorten the ranges of any that overlap with following sequence points 
+        // sort the sequence points back into offset order 
+        for i = 0 to Array.length allSps - 2 do
+            let n,sp1 = allSps.[i]
+            let _,sp2 = allSps.[i+1]
+            if (sp1.Document = sp2.Document) && 
```

Comment:
Seen in other place,  maybe it warrants a function in SequencePoint module or a method in the type, maybe `endOverlapsWithPosition`?